src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -0,0 +1,2261 @@
+			return $this->html;
+		}
+
+		/*
+		 * If there are no updates left to apply, but some have already
+		 * been applied, then finish by copying the rest of the input
+		 * to the end of the updated document and return.
+		 */
+		if ( $requires_no_updating && $this->bytes_already_copied > 0 ) {
+			return $this->output_buffer . substr( $this->html, $this->bytes_already_copied );
+		}
+
+		// Apply the updates, rewind to before the current tag, and reparse the attributes.
+		$content_up_to_opened_tag_name = $this->output_buffer . substr(
+			$this->html,
+			$this->bytes_already_copied,
+			$this->tag_name_starts_at + $this->tag_name_length - $this->bytes_already_copied
+		);
+
+		/*
+		 * 1. Apply the edits by flushing them to the output buffer and updating the copied byte count.
+		 *
+		 * Note: `apply_attributes_updates()` modifies `$this->output_buffer`.
+		 */
+		$this->class_name_updates_to_attributes_updates();
+		$this->apply_attributes_updates();
+
+		/*
+		 * 2. Replace the original HTML with the now-updated HTML so that it's possible to
+		 *    seek to a previous location and have a consistent view of the updated document.
+		 */
+		$this->html                 = $this->output_buffer . substr( $this->html, $this->bytes_already_copied );
+		$this->output_buffer        = $content_up_to_opened_tag_name;
+		$this->bytes_already_copied = strlen( $this->output_buffer );
+
+		/*
+		 * 3. Point this tag processor at the original tag opener and consume it
+		 *
+		 * At this point the internal cursor points to the end of the tag name.
+		 * Rewind before the tag name starts so that it's as if the cursor didn't
+		 * move; a call to `next_tag()` will reparse the recently-updated attributes
+		 * and additional calls to modify the attributes will apply at this same
+		 * location.
+		 *
+		 * <p>Previous HTML<em>More HTML</em></p>
+		 *                 ^  | back up by the length of the tag name plus the opening <
+		 *                 \<-/ back up by strlen("em") + 1 ==> 3
+		 */
+		$this->bytes_already_parsed = strlen( $content_up_to_opened_tag_name ) - $this->tag_name_length - 1;
+		$this->next_tag();
+
+		return $this->html;
+	}
+
+	/**
+	 * Parses tag query input into internal search criteria.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param array|string|null $query {
+	 *     Optional. Which tag name to find, having which class, etc. Default is to find any tag.
+	 *
+	 *     @type string|null $tag_name     Which tag to find, or `null` for "any tag."
+	 *     @type int|null    $match_offset Find the Nth tag matching all search criteria.
+	 *                                     0 for "first" tag, 2 for "third," etc.
+	 *                                     Defaults to first tag.
+	 *     @type string|null $class_name   Tag must contain this class name to match.
+	 *     @type string      $tag_closers  "visit" or "skip": whether to stop on tag closers, e.g. </div>.
+	 * }
+	 *
+	 * @return void
```

Comment:
```suggestion
	 * @return void
```

No empty line b/w params and returns in wpcs. sad clown.

Reply:
🤡 disappointed in 4a438505ec